server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -128,7 +134,8 @@ export class AssetService {
 
       // handle duplicates with a success response
       if (error instanceof QueryFailedError && (error as any).constraint === 'UQ_userid_checksum') {
-        const duplicate = await this.getAssetByChecksum(authUser.id, file.checksum);
+        const checksums = [file.checksum, livePhotoFile?.checksum].filter((checksum): checksum is Buffer => !!checksum);
+        const [duplicate] = await this._assetRepository.getAssetsByChecksums(authUser.id, checksums);
```

Comment:
When uploading a live photo, the duplicate could be either `file` or `livePhotoFile` because the return order is undefined, no?

Reply:
There is an existing bug here, in the scenario where the live photo is a duplicate and the still is not uploaded yet. This code used to look up the still asset by checksum and return it, but since it didn't exist it would throw an EntityNotFound and that asset would fail to backup leading to some sync problem on the mobile app. It's unclear to me what should happen in that situation. I think returning the id to either part is fine. Really, I think, they should be two separate upload requests, which would also solve the issue. That requires mobile app changes and does not seem to be a priority for the mobile devs.

Reply:
Thanks, Jason. I think we can resolve this in this PR and track the issue somewhere else

Reply:
Sounds good. I think this does semi-resolve an existing bug too btw.